sds_data_manager/lambda_code/SDSCode/indexer.py

```diff
@@ -20,16 +20,12 @@
 
 
 def _load_allowed_filenames():
-    # Rather than storing the configuration locally,
-    # we should store the configuration somewhere where things
-    # can be changed on the fly.
-    # For example, a dynamodb table or a section in opensearch
-    current_dir = os.path.dirname(__file__)
-    config_file = os.path.join(current_dir, "config.json")
-
-    with open(config_file) as f:
-        data = json.load(f)
-    return data
+    # get the config file from the S3 bucket
+    config_object = s3.get_object(
```

Comment:
Will this fail if the configuration isn't in S3? Is that ever expected to happen, or does the step below which deploys config.json always run first?

Reply:
Yeah, this will fail if the file isn't there, but the file gets pulled into the S3 bucket during the stack creation. Although I'm now realizing that since the S3 BucketDeployment method that uploads the file only takes directories as inputs, it wouldn't error if the config.json isn't in the config folder. Would it be considered bad practice to have a check for whether config.json exists and some error handling in the stack init?

Reply:
You could check for an empty bucket in the stack check... if the file is required/expected, I think having it fail on deploy or just after deploy is the right way to go. So adding some handling seems reasonable to me, although I would keep it lightweight.

Reply:
It seems like the stack runs through all the code before actually creating any of the resources, so I couldn't check if the bucket is empty in the stack. I'm now checking if that config folder contains a config.json file and raising an error if it doesn't.